src/Core/Streams/FixedSizeStream.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace AsyncAws\Core\Streams;
+
+use AsyncAws\Core\Exception\InvalidArgument;
+
+/**
+ * A Stream decorator that return Chunk with the same exact size.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class FixedSizeStream implements Stream
+{
+    private $decorated;
+
+    private $chunkSize;
+
+    public function __construct(Stream $decorated, int $chunkSize = 64 * 1024)
+    {
+        $this->decorated = $decorated;
+        $this->chunkSize = $chunkSize;
+    }
+
+    public function length(): ?int
+    {
+        return $this->decorated->length();
+    }
+
+    public function stringify(): string
+    {
+        return $this->decorated->stringify();
+    }
+
+    public function getIterator(): \Traversable
```

Comment:
Nice. Did you write this yourself?  

It looks like one of these method you find as a 1000+ votes on StackOverflow  =)

Reply:
thanks for the kind words.

I take inspiration from https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpClient/CurlHttpClient.php#L438-L453

But my version take care of memory